From 444d1183cc9f8ef832050fcc8de1c6ed70a57c2b Mon Sep 17 00:00:00 2001
From: Vignesh Lal T D <vdesikachary@maxlinear.com>
Date: Thu, 22 Aug 2024 12:57:59 +0530
Subject: [PATCH] WLANRTSYS-84203 Hostap log level to debug for missing mld id
 message

Issue: Hostap parse probe request and floods hostap log message with "Invalid
MLO IE is received" and "invalid mld id=-1". From sniffer logs also confirms
that Qualcomm station sends probe request without mld id information in
Multilink tag.

Fix: Changed the log level for these messages to debug (MSG_DEBUG)

FIXUP to WLANRTSYS-63078 Multi BSS changes for MLO
---
 src/ap/mxl_beacon.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ap/mxl_beacon.c b/src/ap/mxl_beacon.c
index ad4851060..9948c5078 100644
--- a/src/ap/mxl_beacon.c
+++ b/src/ap/mxl_beacon.c
@@ -145,7 +145,7 @@ int mxl_hostap_ml_get_non_tx_bss_idx_from_mlid(struct hostapd_data *hapd,  const
 	mld_id = mxl_get_ml_probe_type_mld_id(probe_req_mle, probe_req_mle_len);
 
 	if (mld_id < 0) {
-		wpa_printf(MSG_ERROR, "invalid mld id=%d in mxl_hostap_ml_get_vap_from_mlid!",
+		wpa_printf(MSG_DEBUG, "invalid mld id=%d in mxl_hostap_ml_get_non_tx_bss_idx_from_mlid!",
 			   mld_id);
 		return -1;
 	}
@@ -169,7 +169,7 @@ struct hostapd_data*  mxl_hostap_ml_probe_req_address_check(struct hostapd_data
 	non_tx_bss_idx = mxl_hostap_ml_get_non_tx_bss_idx_from_mlid(hapd, elems->probe_req_mle,
 								    elems->probe_req_mle_len);
 	if (non_tx_bss_idx <= 0) {
-		wpa_printf(MSG_ERROR, "Invalid MLO IE is received from sta " MACSTR,
+		wpa_printf(MSG_DEBUG, "Invalid MLO IE is received from sta " MACSTR,
 				       MAC2STR(mgmt->sa));
 		return NULL;
 	}
-- 
2.43.0

